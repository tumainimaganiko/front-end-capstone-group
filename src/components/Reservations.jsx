import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchReservations } from '../redux/reservations/reservationSlice';

const Reservations = () => {
  const { reservations } = useSelector((state) => state.reservations.reservations);
  const { error } = useSelector((state) => state.reservations);
  const { cars } = useSelector((state) => state.cars);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchReservations());
  }, [dispatch]);

  const { reservationIsLoading } = useSelector((state) => state.reservations);

  return (
    <section className="bg-[#95BF02] fixed top-0 overflow-auto left-0 h-full w-full p-2">
      <h1 className="text-center">Reservations</h1>
      <p className="text-center">{reservationIsLoading ? 'Fetching Reservations ....' : ''}</p>

      {error.length > 0 ? (
        <p className="text-center">
          Error:
          {error}
        </p>
      ) : (
        <table className="mx-auto table-auto border-collapse border border-slate-500">
          <thead>
            <tr>
              <th className="border border-slate-600">Start Date</th>
              <th className="border border-slate-600">End Date</th>
              <th className="border border-slate-600">City</th>
              <th className="border border-slate-600">Vehicles</th>
            </tr>
          </thead>
          <tbody>
            {reservations && Array.isArray(reservations) ? (
              reservations.map((reservation) => (
                <tr key={reservation.id}>
                  <td className="border border-slate-600">{reservation.rental_date}</td>
                  <td className="border border-slate-600">{reservation.date_return}</td>
                  <td className="border border-slate-600">{reservation.destination}</td>
                  <td className="border border-slate-600">
                    {cars.find((car) => car.id === reservation.car_id)?.name}
                  </td>
                </tr>
              ))
            ) : (
              <tr>
                <td colSpan="4" className="border border-slate-600 text-center">No reservations available</td>
              </tr>
            )}
          </tbody>
        </table>
      )}
    </section>
  );
};

export default Reservations;
