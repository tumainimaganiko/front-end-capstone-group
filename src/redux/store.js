import { configureStore } from '@reduxjs/toolkit';
import reservationReducer from './reservations/reservationSlice';
import carsReducer from './cars/carsSlice';
import modelsReducer from './cars/modelsSlice';

const store = configureStore({
  reducer: {
    reservations: reservationReducer,
    cars: carsReducer,
    models: modelsReducer,
  },
});

export default store;
