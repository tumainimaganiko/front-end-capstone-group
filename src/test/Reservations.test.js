import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Reservations from '../components/Reservations';

it('Render a DeleteCar', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Reservations />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
