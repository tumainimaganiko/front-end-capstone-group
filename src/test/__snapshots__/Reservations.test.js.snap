// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a DeleteCar 1`] = `
<section
  className="bg-[#95BF02] fixed top-0 overflow-auto left-0 h-full w-full p-2"
>
  <h1
    className="text-center"
  >
    Reservations
  </h1>
  <p
    className="text-center"
  />
  <table
    className="mx-auto table-auto border-collapse border border-slate-500"
  >
    <thead>
      <tr>
        <th
          className="border border-slate-600"
        >
          Start Date
        </th>
        <th
          className="border border-slate-600"
        >
          End Date
        </th>
        <th
          className="border border-slate-600"
        >
          City
        </th>
        <th
          className="border border-slate-600"
        >
          Vehicles
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
